src/webapp/src/characters/charactersSlice.ts

```diff
@@ -0,0 +1,83 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+
+interface ApiExtraInformation {
+  count: number;
+  pages: number;
+  next: string;
+  prev: string | null;
+}
+
+interface Character {
+  id: number;
+  name: string;
+  status: string;
+  species: string;
+  type: string;
+  gender: string;
+  origin: {
+    name: string;
+    url: string;
+  };
+  location: {
+    name: string;
+    url: string;
+  };
+  episode: string[];
+  url: string;
+  created: Date;
+  image: string;
+}
+
+export interface ApiData {
+  results: Character[];
+  info: ApiExtraInformation;
+  error?: string | null;
+}
+
+const initialState = {
+  results: [] as Character[],
+  info: {} as ApiExtraInformation,
+  error: null,
+} as ApiData;
+
+interface ApiKnownError {
+  error: string;
+}
+
+export const getCharacters = createAsyncThunk<
+  ApiData,
+  void,
+  { rejectValue: ApiKnownError }
+>('characters/getCharacters', async (_, { rejectWithValue }) => {
+  const response = await fetch(`https://rickandmortyapi.com/api/character`);
+
+  if (response.status === 404) {
+    return rejectWithValue((await response.json()) as ApiKnownError);
+  }
+
+  return (await response.json()) as ApiData;
+});
```

Comment:
now what is you have a 500 error does that mean the promise has fulfilled? Most likely no. 

Try using the `response.ok` to check if the response if okay and in _any_ other case we can reject.

Reply:
I see. I just checked against 404 because I used the `rejectWithValue()` function to reject with the specific error I know comes from the API in this case. Per the documentation: 

> If you are performing a request that you know will typically either be a success or have an expected error format, you can pass in a type to rejectValue and return rejectWithValue(knownPayload) in the action creator. This allows you to reference the error payload in the reducer as well as in a component after dispatching the createAsyncThunk action.

So my idea was to reject with the *known* error in case we have a 404.

In the other cases `createAsyncThunk` function is going to: 

> return a standard Redux thunk action creator. The thunk action creator function will have plain action creators for the pending, fulfilled, and rejected cases attached as nested fields.

So because `createAsyncThunk` automatically > Returns a fulfilled promise containing the final dispatched action (either the fulfilled or rejected action object).

I though to just handle the 404 with the `rejectWithValue` function and in the other cases let `creatAsyncThunk` pass the error to `getCharacters.rejected`